core/src/builtin/types.rs

```diff
@@ -7,9 +7,30 @@ use crate as tir_core;
 
 use crate::builtin::DIALECT_NAME;
 
-dialect_type!(FuncType);
-dialect_type!(VoidType);
-dialect_type!(IntType);
+dialect_type!((FuncType, true));
+dialect_type!((VoidType, true));
+dialect_type!((IntType, false));
```

Comment:
Is it possible to make custom assembly optional in the macro syntax and also use a more relevant name than generic true/false?

Reply:
This PR can give you an idea how to parse custom syntax in the macro: https://github.com/perf-toolbox/tir/pull/57